projects/plugins/protect/src/js/hooks/use-protect-data/index.ts

```diff
@@ -140,7 +140,7 @@ export default function useProtectData(
 
 		// Core data may be either a single object or an array of multiple objects.
 		let cores = Array.isArray( data.core ) ? data.core : [];
-		if ( data?.core?.threats ) {
+		if ( data?.core && 'threats' in data.core && data.core?.threats ) {
```

Comment:
Unless I'm missing something, I can't think of a scenario where these two are different...ultimately they both look for a truthy `data.core.threats`.

Reply:
Yeah, but TypeScript doesn't like to access a property if it doesn't exist on that type. Adding those checks asserts that those properties actually exist on those types.

Reply:
That's annoyingly less readable, but so be it. Is the final optional chaining necessary?

```suggestion
		if ( data?.core && 'threats' in data.core && data.core.threats ) {
```

Reply:
No, that's not necessary.